diff -ru tools/paraview/src/paraview/ParaViewCore/ClientServerCore/Core/vtkPVSystemConfigInformation.h crossxlc2/paraview/src/paraview/ParaViewCore/ClientServerCore/Core/vtkPVSystemConfigInformation.h
--- tools/paraview/src/paraview/ParaViewCore/ClientServerCore/Core/vtkPVSystemConfigInformation.h	2013-06-19 13:56:08.000000000 +0000
+++ crossxlc2/paraview/src/paraview/ParaViewCore/ClientServerCore/Core/vtkPVSystemConfigInformation.h	2013-09-11 15:43:26.769328000 +0000
@@ -114,2 +114,2 @@
   vtkPVSystemConfigInformation();
   ~vtkPVSystemConfigInformation();

-private:
+public:
   //BTX
   vector<ConfigInfo> Configs;
   //ETX
diff -ru tools/paraview/src/paraview/ParaViewCore/VTKExtensions/Rendering/CMakeLists.txt crossxlc2/paraview/src/paraview/ParaViewCore/VTKExtensions/Rendering/CMakeLists.txt
--- tools/paraview/src/paraview/ParaViewCore/VTKExtensions/Rendering/CMakeLists.txt	2013-06-19 13:56:09.000000000 +0000
+++ crossxlc2/paraview/src/paraview/ParaViewCore/VTKExtensions/Rendering/CMakeLists.txt	2013-09-11 15:43:26.774732000 +0000
@@ -137,2 +137,2 @@
 endforeach()
 #---------------------------------------------------------

+
+include_directories(${OPENGL_INCLUDE_DIR})
+
 vtk_module_library(vtkPVVTKExtensionsRendering ${Module_SRCS})
diff -ru tools/paraview/src/paraview/VTK/Filters/Geometry/vtkStructuredGridConnectivity.h crossxlc2/paraview/src/paraview/VTK/Filters/Geometry/vtkStructuredGridConnectivity.h
--- tools/paraview/src/paraview/VTK/Filters/Geometry/vtkStructuredGridConnectivity.h	2013-06-19 13:57:32.000000000 +0000
+++ crossxlc2/paraview/src/paraview/VTK/Filters/Geometry/vtkStructuredGridConnectivity.h	2013-09-11 18:09:25.516893042 +0000
@@ -42,5 +42,7 @@
 #include <iostream> // For cout
 #include <vector>   // For STL vector
 #include <map>      // For STL map
+#ifndef __WRAP__
 #include <utility>  // For STL pair and overloaded relational operators
+#endif
 #include <cassert>  // For assert()

 // Forward Declarations
diff -ru tools/paraview/src/paraview/VTK/IO/AMR/vtkAMRBaseReader.h crossxlc2/paraview/src/paraview/VTK/IO/AMR/vtkAMRBaseReader.h
--- tools/paraview/src/paraview/VTK/IO/AMR/vtkAMRBaseReader.h	2013-06-19 13:57:32.000000000 +0000
+++ crossxlc2/paraview/src/paraview/VTK/IO/AMR/vtkAMRBaseReader.h	2013-09-11 18:15:47.947238907 +0000
@@ -24,4 +24,6 @@
 #include "vtkOverlappingAMRAlgorithm.h"
 #include <vector>    // STL vector header
 #include <map>       // STL map header
+#ifndef __WRAP__
 #include <utility>   // for STL pair
+#endif

 // Forward Declarations
 class vtkOverlappingAMR;
