diff -Nr -U5 silo-4.10.2-bsd/SiloWindows/MSVC2012/pdb_detect/pdb_detect.vcxproj silo-4.10.2-bsd.no-perl/SiloWindows/MSVC2012/pdb_detect/pdb_detect.vcxproj
--- silo-4.10.2-bsd/SiloWindows/MSVC2012/pdb_detect/pdb_detect.vcxproj	2014-10-13 20:22:34.000000000 -0400
+++ silo-4.10.2-bsd.no-perl/SiloWindows/MSVC2012/pdb_detect/pdb_detect.vcxproj	2017-04-10 13:39:15.059492963 -0400
@@ -76,12 +76,10 @@
       <TargetMachine>MachineX86</TargetMachine>
     </Link>
     <PostBuildEvent>
       <Message>Create pdbform.h, lite_pdb.h, lite_score.h</Message>
       <Command>$(OutDir)pdb_detect.exe &gt; ..\..\..\src\pdb\pdform.h
-               perl.exe $(SolutionDir)..\..\config\mklite $(SolutionDir)..\..\src\pdb\pdb.h $(SolutionDir)\..\include\lite_pdb.h
-               perl.exe $(SolutionDir)..\..\config\mklite $(SolutionDir)..\..\src\score\score.h $(SolutionDir)\..\include\lite_score.h
       </Command>
     </PostBuildEvent>
     <PreBuildEvent>
       <Command>cd ..\..\
                copysilo.bat
@@ -112,12 +110,10 @@
       <TargetMachine>MachineX64</TargetMachine>
     </Link>
     <PostBuildEvent>
       <Message>Create pdbform.h, lite_pdb.h, lite_score.h</Message>
       <Command>$(OutDir)pdb_detect.exe &gt; ..\..\..\src\pdb\pdform.h
-               perl.exe $(SolutionDir)..\..\config\mklite $(SolutionDir)..\..\src\pdb\pdb.h $(SolutionDir)\..\include\lite_pdb.h
-               perl.exe $(SolutionDir)..\..\config\mklite $(SolutionDir)..\..\src\score\score.h $(SolutionDir)\..\include\lite_score.h
       </Command>
     </PostBuildEvent>
     <PreBuildEvent>
       <Command>cd ..\..\
                copysilo.bat
