diff -Nr -U5 silo-4.10.2-bsd/SiloWindows/MSVC2012/HDF5_Defines.props silo-4.10.2-bsd.hdf5-deplibs/SiloWindows/MSVC2012/HDF5_Defines.props
--- silo-4.10.2-bsd/SiloWindows/MSVC2012/HDF5_Defines.props	2014-10-13 20:22:34.000000000 -0400
+++ silo-4.10.2-bsd.hdf5-deplibs/SiloWindows/MSVC2012/HDF5_Defines.props	2017-04-10 14:03:20.652692007 -0400
@@ -4,16 +4,16 @@
     <_ProjectFileVersion>10.0.40219.1</_ProjectFileVersion>
     <_PropertySheetDisplayName>HDF5_DEFINES</_PropertySheetDisplayName>
   </PropertyGroup>
   <ItemDefinitionGroup>
     <ClCompile>
-      <PreprocessorDefinitions>HAVE_HDF5_DRIVER=1;HAVE_HDF5_H=1;HAVE_LIBHDF5=1;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <PreprocessorDefinitions>HAVE_HDF5_DRIVER=1;HAVE_HDF5_H=1;HAVE_LIBHDF5=1;H5_BUILT_AS_DYNAMIC_LIB=1;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <AdditionalIncludeDirectories Condition="'$(Platform)'=='Win32'">$(HDF5_INC_DIR);$(ZLIB_INC_DIR);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
       <AdditionalIncludeDirectories Condition="'$(Platform)'=='x64'">$(HDF5_INC_DIR_X64);$(ZLIB_INC_DIR);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
       <AdditionalLibraryDirectories Condition="'$(Platform)'=='Win32'">$(HDF5_LIB_DIR);$(ZLIB_LIB_DIR);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
       <AdditionalLibraryDirectories Condition="'$(Platform)'=='x64'">$(HDF5_LIB_DIR_X64);$(ZLIB_LIB_DIR_X64);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
-      <AdditionalDependencies>hdf5.lib;zlib.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>hdf5.lib;zlib.lib;szip.lib;%(AdditionalDependencies)</AdditionalDependencies>
     </Link>
   </ItemDefinitionGroup>
 </Project>
