diff -u -U5 -r silo-4.10.2-bsd/src/hdf5_drv/H5FDsilo.c silo-4.10.2-bsd.hdf5-1.10/src/hdf5_drv/H5FDsilo.c
--- silo-4.10.2-bsd/src/hdf5_drv/H5FDsilo.c	2014-10-13 20:22:32.000000000 -0400
+++ silo-4.10.2-bsd.hdf5-1.10/src/hdf5_drv/H5FDsilo.c	2019-03-18 09:35:48.525227917 -0400
@@ -498,10 +498,11 @@
 
 static const H5FD_class_t H5FD_silo_g = {
     "silo",				        /*name			*/
     MAXADDR,				        /*maxaddr		*/
     H5F_CLOSE_WEAK,				/* fc_degree		*/
+    NULL,                                       /*terminate             */
     H5FD_silo_sb_size,                          /*sb_size               */
     H5FD_silo_sb_encode,                        /*sb_encode             */
     H5FD_silo_sb_decode,                        /*sb_decode             */
     0, 						/*fapl_size		*/
     NULL,					/*fapl_get		*/
