diff --git a/CMakeLists.txt b/CMakeLists.txt
index b6954fb..0cf14f3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -313,6 +313,10 @@ ELSE()
   ENDIF ()
 ENDIF()
 
+IF(CMAKE_CXX_COMPILER_WRAPPER STREQUAL "CrayPrgEnv")
+  SET(EMBREE_ISA_SSE42 OFF)
+ENDIF()
+
 ##############################################################
 # ISA configuration continued
 ##############################################################
diff --git a/common/cmake/crayprgenv.cmake b/common/cmake/crayprgenv.cmake
index c466330..11cc275 100644
--- a/common/cmake/crayprgenv.cmake
+++ b/common/cmake/crayprgenv.cmake
@@ -20,7 +20,5 @@ SET(FLAGS_AVX2      "-target-cpu=haswell")
 SET(FLAGS_AVX512KNL "-target-cpu=mic-knl")
 SET(FLAGS_AVX512SKX "-target-cpu=x86-skylake")
 
-SET_PROPERTY(CACHE EMBREE_ISA_SSE42 PROPERTY VALUE OFF)
-
 STRING(TOLOWER "${CMAKE_CXX_COMPILER_ID}" _lower_compiler_id)
 INCLUDE("${CMAKE_CURRENT_LIST_DIR}/${_lower_compiler_id}.cmake" OPTIONAL)
