diff -U 3 -r src/frmts/gtiff/gt_citation.cpp src-mod/frmts/gtiff/gt_citation.cpp
--- src/frmts/gtiff/gt_citation.cpp	2016-12-08 12:42:59.000000000 -0500
+++ src-mod/frmts/gtiff/gt_citation.cpp	2017-09-25 23:46:10.000000000 -0400
@@ -155,7 +155,7 @@
                 if(p1[0] == '\0' || p1[0] == '\n' || p1[0] == ' ')
                     p1 --;
                 p2 = p1 - 1;
-                while( p2>0 && (p2[0] == ' ' || p2[0] == '\0' || p2[0] == '\n') )
+                while( p2 && (p2[0] == ' ' || p2[0] == '\0' || p2[0] == '\n') )
                     p2--;
                 if(p2 != p1 - 1)
                     p1 = p2;
@@ -198,7 +198,7 @@
                 if(p1[0] == '\0' || p1[0] == '\n' || p1[0] == ' ')
                     p1 --;
                 char* p2 = p1 - 1;
-                while( p2>0 && (p2[0] == ' ' || p2[0] == '\0' || p2[0] == '\n') )
+                while( p2 && (p2[0] == ' ' || p2[0] == '\0' || p2[0] == '\n') )
                     p2--;
                 if(p2 != p1 - 1)
                     p1 = p2;
