diff -u -U5 -r h5py-3.9.0/h5py/h5fd.pyx h5py.noexcept-casts/h5py/h5fd.pyx
--- h5py-3.9.0/h5py/h5fd.pyx	2023-01-10 12:06:37.000000000 -0500
+++ h5py.noexcept-casts/h5py/h5fd.pyx	2023-09-05 21:07:57.948471625 -0400
@@ -193,22 +193,22 @@
 
 info.name = 'fileobj'
 info.maxaddr = libc.stdint.SIZE_MAX - 1
 info.fc_degree = H5F_CLOSE_WEAK
 info.fapl_size = sizeof(PyObject *)
-info.fapl_get = <void *(*)(H5FD_t *)>H5FD_fileobj_fapl_get
-info.fapl_copy = <void *(*)(const void *)>H5FD_fileobj_fapl_copy
-info.fapl_free = <herr_t (*)(void *)>H5FD_fileobj_fapl_free
-info.open = <H5FD_t *(*)(const char *name, unsigned flags, hid_t fapl, haddr_t maxaddr)>H5FD_fileobj_open
-info.close = <herr_t (*)(H5FD_t *)>H5FD_fileobj_close
-info.get_eoa = <haddr_t (*)(const H5FD_t *, H5FD_mem_t)>H5FD_fileobj_get_eoa
-info.set_eoa = <herr_t (*)(H5FD_t *, H5FD_mem_t, haddr_t)>H5FD_fileobj_set_eoa
-info.get_eof = <haddr_t (*)(const H5FD_t *, H5FD_mem_t)>H5FD_fileobj_get_eof
-info.read = <herr_t (*)(H5FD_t *, H5FD_mem_t, hid_t, haddr_t, size_t, void *)>H5FD_fileobj_read
-info.write = <herr_t (*)(H5FD_t *, H5FD_mem_t, hid_t, haddr_t, size_t, const void *)>H5FD_fileobj_write
-info.truncate = <herr_t (*)(H5FD_t *, hid_t, hbool_t)>H5FD_fileobj_truncate
-info.flush = <herr_t (*)(H5FD_t *, hid_t, hbool_t)>H5FD_fileobj_flush
+info.fapl_get = <void *(*)(H5FD_t *) noexcept>H5FD_fileobj_fapl_get
+info.fapl_copy = <void *(*)(const void *) noexcept>H5FD_fileobj_fapl_copy
+info.fapl_free = <herr_t (*)(void *) noexcept>H5FD_fileobj_fapl_free
+info.open = <H5FD_t *(*)(const char *name, unsigned flags, hid_t fapl, haddr_t maxaddr) noexcept>H5FD_fileobj_open
+info.close = <herr_t (*)(H5FD_t *) noexcept>H5FD_fileobj_close
+info.get_eoa = <haddr_t (*)(const H5FD_t *, H5FD_mem_t) noexcept>H5FD_fileobj_get_eoa
+info.set_eoa = <herr_t (*)(H5FD_t *, H5FD_mem_t, haddr_t) noexcept>H5FD_fileobj_set_eoa
+info.get_eof = <haddr_t (*)(const H5FD_t *, H5FD_mem_t) noexcept>H5FD_fileobj_get_eof
+info.read = <herr_t (*)(H5FD_t *, H5FD_mem_t, hid_t, haddr_t, size_t, void *) noexcept>H5FD_fileobj_read
+info.write = <herr_t (*)(H5FD_t *, H5FD_mem_t, hid_t, haddr_t, size_t, const void *) noexcept>H5FD_fileobj_write
+info.truncate = <herr_t (*)(H5FD_t *, hid_t, hbool_t) noexcept>H5FD_fileobj_truncate
+info.flush = <herr_t (*)(H5FD_t *, hid_t, hbool_t) noexcept>H5FD_fileobj_flush
 # H5FD_FLMAP_DICHOTOMY
 info.fl_map = [H5FD_MEM_SUPER,  # default
                H5FD_MEM_SUPER,  # super
                H5FD_MEM_SUPER,  # btree
                H5FD_MEM_DRAW,   # draw
