diff -u -Nru5 matplotlib-3.2.1/setup.py matplotlib-3.2.1.no-jquery/setup.py
--- matplotlib-3.2.1/setup.py	2020-03-18 00:07:16.000000000 -0400
+++ matplotlib-3.2.1.no-jquery/setup.py	2020-03-23 10:07:41.809005503 -0400
@@ -144,26 +144,26 @@
 
 # Relying on versioneer's implementation detail.
 class sdist_with_jquery(cmdclass['sdist']):
     def make_release_tree(self, base_dir, files):
         super().make_release_tree(base_dir, files)
-        _download_jquery_to(
-            Path(base_dir, "lib/matplotlib/backends/web_backend/"))
+        # _download_jquery_to(
+        #     Path(base_dir, "lib/matplotlib/backends/web_backend/"))
 
 
 # Affects install and bdist_wheel.
 class install_lib_with_jquery(InstallLibCommand):
     def run(self):
         super().run()
-        _download_jquery_to(
-            Path(self.install_dir, "matplotlib/backends/web_backend/"))
+        # _download_jquery_to(
+        #     Path(self.install_dir, "matplotlib/backends/web_backend/"))
 
 
 class develop_with_jquery(DevelopCommand):
     def run(self):
         super().run()
-        _download_jquery_to(Path("lib/matplotlib/backends/web_backend/"))
+        # _download_jquery_to(Path("lib/matplotlib/backends/web_backend/"))
 
 
 cmdclass['sdist'] = sdist_with_jquery
 cmdclass['install_lib'] = install_lib_with_jquery
 cmdclass['develop'] = develop_with_jquery
