From 4433a7404973f865abb721e5b57f9b48f7ee3013 Mon Sep 17 00:00:00 2001
From: Mathieu Westphal <mathieu.westphal@kitware.com>
Date: Fri, 10 Feb 2023 15:53:51 +0100
Subject: [PATCH] Remove unused boost component

---
 cmake/FindSalomeBoost.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/FindSalomeBoost.cmake b/cmake/FindSalomeBoost.cmake
index c80f02d..2c2e0e7 100644
--- a/cmake/FindSalomeBoost.cmake
+++ b/cmake/FindSalomeBoost.cmake
@@ -27,7 +27,7 @@ SET(Boost_USE_STATIC_LIBS        OFF)
 SET(Boost_USE_MULTITHREADED      ON)
 SET(Boost_USE_STATIC_RUNTIME     OFF)
 SET(Boost_NO_BOOST_CMAKE         ON)
-SET(SalomeBoost_COMPONENTS       filesystem regex system thread date_time chrono serialization)
+SET(SalomeBoost_COMPONENTS       system)
 SET(SalomeBoost_FIND_COMPONENTS ${SalomeBoost_COMPONENTS})
 
 IF(WIN32)
-- 
2.39.1

