diff --git a/openvkl/common/VKLCommon.cpp b/openvkl/common/VKLCommon.cpp
index a9e2f57..aa240d1 100644
--- a/openvkl/common/VKLCommon.cpp
+++ b/openvkl/common/VKLCommon.cpp
@@ -13,7 +13,7 @@ namespace openvkl {
   VKLError loadLocalModule(const std::string &moduleName)
   {
     std::string libName = "openvkl_module_" + moduleName;
-    rkcommon::loadLibrary(libName);
+    rkcommon::loadLibrary(libName, true);
 
     std::string initSymName = "openvkl_init_module_" + moduleName;
     void *initSym           = rkcommon::getSymbol(initSymName);
