diff --git a/components/ospcommon/cmake/FindTBB.cmake b/components/ospcommon/cmake/FindTBB.cmake
--- a/components/ospcommon/cmake/FindTBB.cmake
+++ b/components/ospcommon/cmake/FindTBB.cmake
@@ -69,14 +69,22 @@
     SET(TBB_VCVER vc14)
   ENDIF()
 
-  SET(TBB_LIBDIR ${TBB_ROOT}/lib/${TBB_ARCH}/${TBB_VCVER})
-  SET(TBB_BINDIR ${TBB_ROOT}/bin/${TBB_ARCH}/${TBB_VCVER})
+  SET(TBB_LIBDIR ${TBB_ROOT}/lib/)
+  SET(TBB_BINDIR ${TBB_ROOT}/bin/)
 
   FIND_PATH(TBB_INCLUDE_DIR tbb/task_scheduler_init.h PATHS ${TBB_ROOT}/include NO_DEFAULT_PATH)
-  FIND_LIBRARY(TBB_LIBRARY tbb PATHS ${TBB_LIBDIR} NO_DEFAULT_PATH)
-  FIND_LIBRARY(TBB_LIBRARY_DEBUG tbb_debug PATHS ${TBB_LIBDIR} NO_DEFAULT_PATH)
-  FIND_LIBRARY(TBB_LIBRARY_MALLOC tbbmalloc PATHS ${TBB_LIBDIR} NO_DEFAULT_PATH)
-  FIND_LIBRARY(TBB_LIBRARY_MALLOC_DEBUG tbbmalloc_debug PATHS ${TBB_LIBDIR} NO_DEFAULT_PATH)
+  FIND_LIBRARY(TBB_LIBRARY tbb
+    PATHS ${TBB_LIBDIR} ${TBB_LIBDIR}/${TBB_ARCH}/${TBB_VCVER}
+    NO_DEFAULT_PATH)
+  FIND_LIBRARY(TBB_LIBRARY_DEBUG tbb_debug
+    PATHS ${TBB_LIBDIR} ${TBB_LIBDIR}/${TBB_ARCH}/${TBB_VCVER}
+    NO_DEFAULT_PATH)
+  FIND_LIBRARY(TBB_LIBRARY_MALLOC tbbmalloc
+    PATHS ${TBB_LIBDIR} ${TBB_LIBDIR}/${TBB_ARCH}/${TBB_VCVER}
+    NO_DEFAULT_PATH)
+  FIND_LIBRARY(TBB_LIBRARY_MALLOC_DEBUG tbbmalloc_debug
+    PATHS ${TBB_LIBDIR} ${TBB_LIBDIR}/${TBB_ARCH}/${TBB_VCVER}
+    NO_DEFAULT_PATH)
 
 ELSE ()
 
diff --git a/ospray/CMakeLists.txt b/ospray/CMakeLists.txt
--- a/ospray/CMakeLists.txt
+++ b/ospray/CMakeLists.txt
@@ -32,6 +32,7 @@
       ${TBB_ROOT}/../redist/${TBB_ARCH}_win/tbb/${TBB_VCVER}
       ${TBB_ROOT}/../redist/${TBB_ARCH}/tbb/${TBB_VCVER}
       ${TBB_ROOT}/bin/${TBB_ARCH}/${TBB_VCVER}
+      ${TBB_ROOT}/bin
     )
     FIND_FILE(TBB_DLL tbb.dll HINTS ${TBB_DLL_HINTS})
     FIND_FILE(TBB_DLL_MALLOC tbbmalloc.dll PATHS HINTS ${TBB_DLL_HINTS})
