diff --git a/cmake/ospray_cmake_config/osprayConfig.cmake.in b/cmake/ospray_cmake_config/osprayConfig.cmake.in
index 2464502..e7bb448 100644
--- a/cmake/ospray_cmake_config/osprayConfig.cmake.in
+++ b/cmake/ospray_cmake_config/osprayConfig.cmake.in
@@ -155,7 +155,9 @@ if(OSPRAY_USE_EXTERNAL_EMBREE)
   # Find existing Embree on the machine
   find_package(embree @EMBREE_VERSION_REQUIRED@ REQUIRED)
 else()
-  if (APPLE)
+  if (WIN32)
+    set(EMBREE_LIBRARY ${LIBRARY_PATH_PREFIX}embree${LIBRARY_SUFFIX})
+  elseif (APPLE)
     set(EMBREE_LIBRARY ${LIBRARY_PATH_PREFIX}embree.2${LIBRARY_SUFFIX})
   else()
     set(EMBREE_LIBRARY ${LIBRARY_PATH_PREFIX}embree${LIBRARY_SUFFIX}.2)
@@ -201,6 +203,10 @@ set(CURRENT_ROOT_INSTALL_DIR   ${OSPRAY_CURRENT_ROOT_INSTALL_DIR})
 # Remove duplicate entries from OSPRAY_INCLUDE_DIRS
 list(REMOVE_DUPLICATES OSPRAY_INCLUDE_DIRS)
 
+set(ospray_libraries
+  ${LIBRARY_PATH_PREFIX}ospray${LIBRARY_SUFFIX}
+  ${LIBRARY_PATH_PREFIX}ospray_common${LIBRARY_SUFFIX})
+
 # Set expected OSPRAY_LIBRARIES variable for FindPackage().
 set(OSPRAY_LIBRARIES
   ${OSPRAY_DEPENDENCIES}
@@ -209,6 +215,7 @@ set(OSPRAY_LIBRARIES
   ${EMBREE_LIBRARY}
   ${LIBRARY_PATH_PREFIX}ospray${LIBRARY_SUFFIX}
   ${LIBRARY_PATH_PREFIX}ospray_common${LIBRARY_SUFFIX}
+  ${ospray_libraries}
 )
 
 # Reset CMake module path to its state when this script was called.
