diff --git a/doc/ospray-doc b/doc/ospray-doc
deleted file mode 160000
index e89eabcc5..000000000
--- a/doc/ospray-doc
+++ /dev/null
@@ -1 +0,0 @@
-Subproject commit e89eabcc590d7ec94be655305f52b5b03addbf8d
diff --git a/ospray/common/OSPCommon.cpp b/ospray/common/OSPCommon.cpp
index 7ff2b34a4..55e16aead 100644
--- a/ospray/common/OSPCommon.cpp
+++ b/ospray/common/OSPCommon.cpp
@@ -582,7 +582,7 @@ uint32_t logLevel()
 OSPError loadLocalModule(const std::string &name)
 {
   std::string libName = "ospray_module_" + name;
-  loadLibrary(libName, false);
+  loadLibrary(libName, true);
 
   std::string initSymName = "ospray_module_init_" + name;
   void *initSym = getSymbol(initSymName);
