diff -u -U5 -r PDAL-2.5.1-src/plugins/e57/libE57Format/include/E57Format.h PDAL-2.5.1-src.cstdint-includes/plugins/e57/libE57Format/include/E57Format.h
--- PDAL-2.5.1-src/plugins/e57/libE57Format/include/E57Format.h	2023-02-10 09:53:57.000000000 -0500
+++ PDAL-2.5.1-src.cstdint-includes/plugins/e57/libE57Format/include/E57Format.h	2023-09-06 12:45:31.017545823 -0400
@@ -30,10 +30,11 @@
  */
 
 //! @file  E57Format.h header file for the E57 API
 
 #include <cfloat>
+#include <cstdint>
 #include <memory>
 #include <vector>
 
 #include "E57Exception.h"
 
