diff --git a/vendor/arbiter/CMakeLists.txt b/vendor/arbiter/CMakeLists.txt
index a7d3f6c7e..fa9970dfa 100644
--- a/vendor/arbiter/CMakeLists.txt
+++ b/vendor/arbiter/CMakeLists.txt
@@ -12,9 +12,15 @@ target_include_directories(${PDAL_ARBITER_LIB_NAME}
     PRIVATE
         ${NLOHMANN_INCLUDE_DIR}
 )
+if (NOT TARGET CURL::libcurl)
+    add_library(CURL::libcurl UNKNOWN IMPORTED)
+    set_target_properties(CURL::libcurl
+    PROPERTIES
+        IMPORTED_LOCATION "${CURL_LIBRARIES}")
+endif ()
 target_link_libraries(${PDAL_ARBITER_LIB_NAME}
     PRIVATE
-        ${CURL_LIBRARIES}
+        CURL::libcurl
 )
 target_compile_definitions(${PDAL_ARBITER_LIB_NAME}
     PRIVATE
