diff --git a/Utilities/WrapClientServer/vtkWrapClientServer.c b/Utilities/WrapClientServer/vtkWrapClientServer.c
index 1ea05dc..dc52f16 100644
--- a/Utilities/WrapClientServer/vtkWrapClientServer.c
+++ b/Utilities/WrapClientServer/vtkWrapClientServer.c
@@ -1114,8 +1114,8 @@ int main(int argc, char *argv[])
   fprintf(fp,"#define VTK_WRAPPING_CXX\n");
   if(strcmp("vtkObjectBase", data->Name) != 0)
     {
-    /* Block inclusion of full streams. */
-    fprintf(fp,"#define VTK_STREAMS_FWD_ONLY\n");
+    ///* Block inclusion of full streams. */
+    //fprintf(fp,"#define VTK_STREAMS_FWD_ONLY\n");
     }
   fprintf(fp,"#include \"vtkSystemIncludes.h\"\n");
   fprintf(fp,"#include \"%s.h\"\n",data->Name);
