diff --git a/IO/Export/CMakeLists.txt b/IO/Export/CMakeLists.txt
index 1c4c049..b12d2ea 100644
--- a/IO/Export/CMakeLists.txt
+++ b/IO/Export/CMakeLists.txt
@@ -30,4 +30,6 @@ set_source_files_properties(
   ABSTRACT
   )
 
+
+include_directories(${OPENGL_INCLUDE_DIR})
 vtk_module_library(vtkIOExport ${Module_SRCS})
diff --git a/IO/NetCDF/vtkMPASReader.cxx b/IO/NetCDF/vtkMPASReader.cxx
index 67f4457..32977cc 100644
--- a/IO/NetCDF/vtkMPASReader.cxx
+++ b/IO/NetCDF/vtkMPASReader.cxx
@@ -2363,7 +2363,9 @@ void vtkMPASReader::PrintSelf(ostream& os, vtkIndent indent)
   this->Superclass::PrintSelf(os, indent);
   os << indent << "FileName: "
      << (this->FileName?this->FileName:"NULL") << "\n";
-  os << indent << "VerticalLevelRange: " << this->VerticalLevelRange << "\n";
+  os << indent << "VerticalLevelRange: " 
+     << this->VerticalLevelRange[0] << ","
+     << this->VerticalLevelRange[1] << "\n";
   os << indent << "this->NumberOfPointVars: " << this->NumberOfPointVars << "\n";
   os << indent << "this->NumberOfCellVars: " << this->NumberOfCellVars << "\n";
   os << indent << "this->MaximumPoints: " << this->MaximumPoints << "\n";
diff --git a/Interaction/Widgets/vtkResliceCursorActor.cxx b/Interaction/Widgets/vtkResliceCursorActor.cxx
index 6783467..d91721d 100644
--- a/Interaction/Widgets/vtkResliceCursorActor.cxx
+++ b/Interaction/Widgets/vtkResliceCursorActor.cxx
@@ -332,8 +332,8 @@ void vtkResliceCursorActor::PrintSelf(ostream& os, vtkIndent indent)
 {
   this->Superclass::PrintSelf(os,indent);
 
-  vtkPrintMemberObjectMacro( CursorCenterlineActor, os, indent );
-  vtkPrintMemberObjectMacro( CursorAlgorithm, os, indent );
+  //vtkPrintMemberObjectMacro( CursorCenterlineActor, os, indent );
+  //vtkPrintMemberObjectMacro( CursorAlgorithm, os, indent );
 
   // this->CursorCenterlineMapper[3];
   // this->CursorCenterlineActor[3];
diff --git a/Rendering/Annotation/vtkAxisActor.cxx b/Rendering/Annotation/vtkAxisActor.cxx
index 289acea..01101a5 100644
--- a/Rendering/Annotation/vtkAxisActor.cxx
+++ b/Rendering/Annotation/vtkAxisActor.cxx
@@ -1130,7 +1130,10 @@ void vtkAxisActor::PrintSelf(ostream& os, vtkIndent indent)
       ;
     }
 
-  os << indent << "DeltaMajor: " << this->DeltaMajor << endl;
+  os << indent << "DeltaMajor: "
+     << this->DeltaMajor[0] << ","
+     << this->DeltaMajor[1] << ","
+     << this->DeltaMajor[2] << endl;
   os << indent << "DeltaMinor: " << this->DeltaMinor << endl;
   os << indent << "DeltaRangeMajor: " << this->DeltaRangeMajor << endl;
   os << indent << "DeltaRangeMinor: " << this->DeltaRangeMinor << endl;
@@ -1164,8 +1167,10 @@ void vtkAxisActor::PrintSelf(ostream& os, vtkIndent indent)
 
   os << indent << "DrawGridlines: " << this->DrawGridlines << endl;
 
-  os << indent << "MajorStart: " << this->MajorStart << endl;
-  os << indent << "MinorStart: " << this->MinorStart << endl;
+  os << indent << "MajorStart: "
+     << this->MajorStart[0] << ","
+     << this->MajorStart[1] << ","
+     << this->MajorStart[2] << endl;
 
   os << indent << "AxisPosition: " << this->AxisPosition << endl;
 
diff --git a/Rendering/Context2D/CMakeLists.txt b/Rendering/Context2D/CMakeLists.txt
index 804269a..36568b0 100644
--- a/Rendering/Context2D/CMakeLists.txt
+++ b/Rendering/Context2D/CMakeLists.txt
@@ -49,4 +49,7 @@ set_source_files_properties(
   WRAP_EXCLUDE
   )
 
+
+include_directories(${OPENGL_INCLUDE_DIR})
+
 vtk_module_library(vtkRenderingContext2D ${module_SRCS})
diff --git a/Rendering/FreeTypeOpenGL/CMakeLists.txt b/Rendering/FreeTypeOpenGL/CMakeLists.txt
index 71a3386..86c1ea2 100644
--- a/Rendering/FreeTypeOpenGL/CMakeLists.txt
+++ b/Rendering/FreeTypeOpenGL/CMakeLists.txt
@@ -32,4 +32,7 @@ set_source_files_properties(
   WRAP_EXCLUDE
   )
 
+
+include_directories(${OPENGL_INCLUDE_DIR})
+
 vtk_module_library(${vtk-module} ${Module_SRCS})
diff --git a/Rendering/GL2PS/CMakeLists.txt b/Rendering/GL2PS/CMakeLists.txt
index c5246df..82cbc2b 100644
--- a/Rendering/GL2PS/CMakeLists.txt
+++ b/Rendering/GL2PS/CMakeLists.txt
@@ -5,4 +5,8 @@ set(Module_SRCS
 set_source_files_properties(
   vtkGL2PSContextDevice2D
   WRAP_EXCLUDE)
+
+
+include_directories(${OPENGL_INCLUDE_DIR})
+
 vtk_module_library(${vtk-module} ${Module_SRCS})
diff --git a/Rendering/HybridOpenGL/CMakeLists.txt b/Rendering/HybridOpenGL/CMakeLists.txt
index 1b72090..0ca2cdc 100644
--- a/Rendering/HybridOpenGL/CMakeLists.txt
+++ b/Rendering/HybridOpenGL/CMakeLists.txt
@@ -31,4 +31,7 @@ foreach(file ${shader_files})
   set_source_files_properties(${file_we} WRAP_EXCLUDE)
 endforeach()
 
+
+include_directories(${OPENGL_INCLUDE_DIR})
+
 vtk_module_library(${vtk-module} ${Module_SRCS})
diff --git a/Rendering/Label/CMakeLists.txt b/Rendering/Label/CMakeLists.txt
index 22ed7e1..f4245c3 100644
--- a/Rendering/Label/CMakeLists.txt
+++ b/Rendering/Label/CMakeLists.txt
@@ -20,5 +20,7 @@ set_source_files_properties(
   ABSTRACT
   )
 
+include_directories(${OPENGL_INCLUDE_DIR})
+
 include_directories(${VTK_SOURCE_DIR}/Utilities) # octree
 vtk_module_library(vtkRenderingLabel ${Module_SRCS})
diff --git a/Rendering/Parallel/CMakeLists.txt b/Rendering/Parallel/CMakeLists.txt
index 4e0d25f..a160b9e 100644
--- a/Rendering/Parallel/CMakeLists.txt
+++ b/Rendering/Parallel/CMakeLists.txt
@@ -46,4 +46,6 @@ set_source_files_properties(
   WRAP_EXCLUDE
   )
 
+include_directories(${OPENGL_INCLUDE_DIR})
+
 vtk_module_library(vtkRenderingParallel ${Module_SRCS})
diff --git a/Rendering/VolumeOpenGL/CMakeLists.txt b/Rendering/VolumeOpenGL/CMakeLists.txt
index 7f8e84a..cf9b8e2 100644
--- a/Rendering/VolumeOpenGL/CMakeLists.txt
+++ b/Rendering/VolumeOpenGL/CMakeLists.txt
@@ -117,4 +117,7 @@ set_source_files_properties(
   WRAP_EXCLUDE
   )
 
+
+include_directories(${OPENGL_INCLUDE_DIR})
+
 vtk_module_library(vtkRenderingVolumeOpenGL ${Module_SRCS})
diff --git a/ThirdParty/gl2ps/vtkgl2ps/CMakeLists.txt b/ThirdParty/gl2ps/vtkgl2ps/CMakeLists.txt
index e8bbd1d..f9564cd 100644
--- a/ThirdParty/gl2ps/vtkgl2ps/CMakeLists.txt
+++ b/ThirdParty/gl2ps/vtkgl2ps/CMakeLists.txt
@@ -27,6 +27,9 @@ include_directories(
   ${GL2PS_GL_INCLUDE_DIR}
 )
 
+
+include_directories(${OPENGL_INCLUDE_DIR})
+
 # Source files
 set(GL2PS_srcs
   gl2ps.c
diff --git a/ThirdParty/sqlite/vtksqlite/vtk_sqlite3.c b/ThirdParty/sqlite/vtksqlite/vtk_sqlite3.c
index 7e6fcac..3684290 100644
--- a/ThirdParty/sqlite/vtksqlite/vtk_sqlite3.c
+++ b/ThirdParty/sqlite/vtksqlite/vtk_sqlite3.c
@@ -369,6 +369,7 @@
 ** Older versions of Vtk_Sqlite used an optional THREADSAFE macro.
 ** We support that for legacy
 */
+#define VTK_SQLITE_THREADSAFE 0
 #if !defined(VTK_SQLITE_THREADSAFE)
 #if defined(THREADSAFE)
 # define VTK_SQLITE_THREADSAFE THREADSAFE
