--- llvm-5.0.1.src/lib/IR/Attributes.cpp.orig	2018-02-14 16:39:55.983961116 -0500
+++ llvm-5.0.1.src/lib/IR/Attributes.cpp	2018-02-14 16:39:15.111672713 -0500
@@ -795,7 +795,7 @@
 static constexpr unsigned attrIdxToArrayIdx(unsigned Index) {
   // MSVC warns about '~0U + 1' wrapping around when this is called on
   // FunctionIndex, so cast to int first.
-  return static_cast<int>(Index) + 1;
+  return static_cast<unsigned>(static_cast<int>(Index)) + 1;
 }
 
 AttributeListImpl::AttributeListImpl(LLVMContext &C,
