diff -Nr -U5 manta-r2439/Core/Containers/Array3.h manta-r2439.fix-missing-includes/Core/Containers/Array3.h
--- manta-r2439/Core/Containers/Array3.h	2010-12-29 11:43:30.000000000 -0500
+++ manta-r2439.fix-missing-includes/Core/Containers/Array3.h	2015-12-02 16:19:30.385774756 -0500
@@ -49,10 +49,14 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <Core/Util/Assert.h>
 
+#if defined(__GNUC__) && defined(__linux)
+#include <unistd.h>
+#endif
+
 namespace Manta {
 
 /**************************************
 
  CLASS
diff -Nr -U5 manta-r2439/Core/Thread/Thread.cc manta-r2439.fix-missing-includes/Core/Thread/Thread.cc
--- manta-r2439/Core/Thread/Thread.cc	2010-12-29 11:43:30.000000000 -0500
+++ manta-r2439.fix-missing-includes/Core/Thread/Thread.cc	2015-12-02 16:19:40.672085924 -0500
@@ -59,10 +59,12 @@
 #  include <strings.h>
 #endif
 
 #include <string.h>
 #include <sys/types.h>
+#include <unistd.h>
+
 #ifdef _WIN32
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
 #include <winnt.h>
 #include <io.h>
