diff -Nr -U5 manta-r2439/Core/CMakeLists.txt manta-r2439.fix-missing-link/Core/CMakeLists.txt
--- manta-r2439/Core/CMakeLists.txt	2010-12-29 11:43:31.000000000 -0500
+++ manta-r2439.fix-missing-link/Core/CMakeLists.txt	2015-12-02 16:19:44.493201515 -0500
@@ -180,10 +180,14 @@
      Util/UpdateGraph.h
      )
 
 ADD_LIBRARY (Manta_Core ${CORE_SOURCES})
 
+IF (UNIX AND NOT APPLE)
+  TARGET_LINK_LIBRARIES(Manta_Core dl)
+ENDIF()
+
 IF(LIBXML2_FOUND)
   TARGET_LINK_LIBRARIES(Manta_Core ${LIBXML2_LIBRARY})
 ENDIF(LIBXML2_FOUND)
 
 IF (CMAKE_USE_PTHREADS_INIT)
