diff --git a/configure.ac b/configure.ac
index c9d64cd..644aa17 100644
--- a/configure.ac
+++ b/configure.ac
@@ -945,7 +945,7 @@ llvm_add_target() {
 # Call this inside ` ` to get the return value.
 # $1 is the llvm-config command with arguments.
 strip_unwanted_llvm_flags() {
-    echo " `$1` " | sed -E \
+    echo " `$1` " | sed -r \
         -e 's/[[[:space:]]]+-m[[^[:space:]]]*//g' \
         -e 's/[[[:space:]]]+-DNDEBUG[[[:space:]]]/ /g' \
         -e 's/[[[:space:]]]+-D_GNU_SOURCE[[[:space:]]]/ /g' \

