diff --git a/configure.ac b/configure.ac
index d215b63..5908dfb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1377,6 +1377,9 @@ AC_ARG_ENABLE([driglx-direct],
 dnl
 dnl libGL configuration per driver
 dnl
+if test "x$enable_glx" != xno; then
+    PKG_CHECK_MODULES([GLPROTO], [glproto >= $GLPROTO_REQUIRED])
+fi
 case "x$enable_glx" in
 xxlib | xgallium-xlib)
     # Xlib-based GLX
@@ -1390,7 +1393,6 @@ xxlib | xgallium-xlib)
     ;;
 xdri)
     # DRI-based GLX
-    PKG_CHECK_MODULES([GLPROTO], [glproto >= $GLPROTO_REQUIRED])
 
     # find the DRI deps for libGL
     dri_modules="x11 xext xdamage xfixes x11-xcb xcb xcb-glx >= $XCBGLX_REQUIRED"
diff --git a/src/gallium/state_trackers/glx/xlib/Makefile.am b/src/gallium/state_trackers/glx/xlib/Makefile.am
index a7e6c0c..112030be 100644
--- a/src/gallium/state_trackers/glx/xlib/Makefile.am
+++ b/src/gallium/state_trackers/glx/xlib/Makefile.am
@@ -25,6 +25,7 @@ include $(top_srcdir)/src/gallium/Automake.inc
 
 AM_CFLAGS = \
 	$(GALLIUM_CFLAGS) \
+	$(GLPROTO_CFLAGS) \
 	$(X11_INCLUDES)
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/include \
