diff -u -Nr -U5 gmsh-4.11.1-source/contrib/blossom/concorde97/EDGEGEN/edgegen.c gmsh-4.11.1-source.string-includes/contrib/blossom/concorde97/EDGEGEN/edgegen.c
--- gmsh-4.11.1-source/contrib/blossom/concorde97/EDGEGEN/edgegen.c	2020-09-08 20:07:54.000000000 +0200
+++ gmsh-4.11.1-source.string-includes/contrib/blossom/concorde97/EDGEGEN/edgegen.c	2025-01-07 22:25:51.959098017 +0100
@@ -87,10 +87,12 @@
 #include "edgegen.h"
 #include "kdtree.h"
 #include "linkern.h"
 #include "fmatch.h"
 
+#include <string.h>
+
 #define SWAP(x,y,t) ((t)=(x),(x)=(y),(y)=(t))
 
 typedef struct intptr {
     int this;
     struct intptr *next;
diff -u -Nr -U5 gmsh-4.11.1-source/contrib/blossom/concorde97/TSP/prob_io.c gmsh-4.11.1-source.string-includes/contrib/blossom/concorde97/TSP/prob_io.c
--- gmsh-4.11.1-source/contrib/blossom/concorde97/TSP/prob_io.c	2020-09-08 20:07:54.000000000 +0200
+++ gmsh-4.11.1-source.string-includes/contrib/blossom/concorde97/TSP/prob_io.c	2025-01-07 22:29:56.246186706 +0100
@@ -69,10 +69,12 @@
 #include "machdefs.h"
 #include "util.h"
 #include "tsp.h"
 #include "bigguy.h"
 
+#include <string.h>
+
 #ifdef CC_PROTOTYPE_ANSI
 
 static int
     prob_name (char *buf, int buflen, char *f, int n),
     write_header (CCtsp_PROB_FILE *p);
diff -u -Nr -U5 gmsh-4.11.1-source/contrib/blossom/concorde97/TSP/tsp_lp.c gmsh-4.11.1-source.string-includes/contrib/blossom/concorde97/TSP/tsp_lp.c
--- gmsh-4.11.1-source/contrib/blossom/concorde97/TSP/tsp_lp.c	2020-09-08 20:07:54.000000000 +0200
+++ gmsh-4.11.1-source.string-includes/contrib/blossom/concorde97/TSP/tsp_lp.c	2025-01-07 22:30:09.744193715 +0100
@@ -216,10 +216,12 @@
 #include "tsp.h"
 #include "lp.h"
 #include "bigguy.h"
 #include "cut.h"
 
+#include <string.h>
+
 /*
    parameters for addbad_variables.  Addbad generates edges (x,y) with
    len(x,y) - pi_ub[x] - pi_ub[y] < 0 in chunks of PRICE_GEN, computes
    rc for that chunk, and adds those with rc(x,y) < CCtsp_PRICE_RCTHRESH to the
    pool.  This continues until the pool contains more than PRICE_POOL edges,
diff -u -Nr -U5 gmsh-4.11.1-source/contrib/blossom/concorde97/UTIL/getdata.c gmsh-4.11.1-source.string-includes/contrib/blossom/concorde97/UTIL/getdata.c
--- gmsh-4.11.1-source/contrib/blossom/concorde97/UTIL/getdata.c	2020-09-08 20:07:54.000000000 +0200
+++ gmsh-4.11.1-source.string-includes/contrib/blossom/concorde97/UTIL/getdata.c	2025-01-08 00:44:09.557293179 +0100
@@ -160,10 +160,12 @@
 /***************************************************************************/
 
 #include "machdefs.h"
 #include "util.h"
 
+#include <string.h>
+
 #define MATRAND_SCALE 10  /* Range of edge lengths: [0, MATRAND_SCALE * n) */
 
 #ifdef CC_PROTOTYPE_ANSI
 
 static void
diff -u -Nr -U5 gmsh-4.11.1-source/contrib/blossom/concorde97/UTIL/safe_io.c gmsh-4.11.1-source.string-includes/contrib/blossom/concorde97/UTIL/safe_io.c
--- gmsh-4.11.1-source/contrib/blossom/concorde97/UTIL/safe_io.c	2020-09-08 20:07:54.000000000 +0200
+++ gmsh-4.11.1-source.string-includes/contrib/blossom/concorde97/UTIL/safe_io.c	2025-01-08 00:45:00.080356133 +0100
@@ -55,10 +55,14 @@
 /***************************************************************************/
 
 #include "machdefs.h"
 #include "util.h"
 
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
 #ifndef SEEK_SET
 #ifdef L_SET
 #define SEEK_SET L_SET
 #else
 #define SEEK_SET 0
