From d87540f07efd2f3ad6740f1e83731e151bc668be Mon Sep 17 00:00:00 2001
From: Cory Quammen <cory.quammen@kitware.com>
Date: Mon, 10 Jun 2019 00:11:26 -0400
Subject: [PATCH] Darwin patch 3

---
 src/mili_internal.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/mili_internal.h b/src/mili_internal.h
index 0fca996..8182410 100755
--- a/src/mili_internal.h
+++ b/src/mili_internal.h
@@ -555,9 +555,9 @@ typedef struct _mili_family
  * Library-private file family management routines and data.
  */
 
-int host_index;
-int internal_sizes[QTY_PD_ENTRY_TYPES + 1];
-int mili_verbose;
+extern int host_index;
+extern int internal_sizes[QTY_PD_ENTRY_TYPES + 1];
+extern int mili_verbose;
 Return_value validate_fam_id( Famid fam_id );
 Return_value parse_control_string( char *ctl_str, Mili_family *fam,
                                    Bool_type *p_create );
@@ -625,7 +625,7 @@ void delete_dir( Mili_family *fam );
 Return_value load_directories( Mili_family *fam );
 
 /* param.c - parameter management routines. */
-char *dtype_names[QTY_PD_ENTRY_TYPES + 1];
+extern char *dtype_names[QTY_PD_ENTRY_TYPES + 1];
 Return_value read_scalar( Mili_family *fam, Param_ref *p_pr,  void *p_value );
 Return_value mili_read_string( Mili_family *fam, Param_ref *p_pr,
                                char *p_value );
@@ -668,7 +668,7 @@ void get_mili_version( char *mili_version_ptr );
 /* dep.c - routines for handling architecture dependencies. */
 Return_value set_default_io_routines( Mili_family *fam );
 Return_value set_state_data_io_routines( Mili_family *fam );
-void (*write_funcs[QTY_PD_ENTRY_TYPES + 1])();
+extern void (*write_funcs[QTY_PD_ENTRY_TYPES + 1])();
 
 /* svar.c - routines for managing state variables. */
 Bool_type valid_svar_data( Aggregate_type atype, char *name,
@@ -761,7 +761,7 @@ Return_value mc_get_class_info_by_index(Mili_family* in, int *mesh_id,
 void mili_delete_mo_class_data( void *p_data );
 
 /* wrap_c.c - C-half of FORTRAN-to-C wrappers. */
-int fortran_api;
+extern int fortran_api;
 /* write_db.c */
 Return_value
 write_state_data( int state_num, Mili_analysis *out_db );
-- 
2.10.1

