From e2562d32e496b31acf4f4f099997af094e0e56ba Mon Sep 17 00:00:00 2001
From: Cory Quammen <cory.quammen@kitware.com>
Date: Mon, 10 Jun 2019 00:08:55 -0400
Subject: [PATCH] Darwin patch 1

---
 src/mesh_u.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mesh_u.c b/src/mesh_u.c
index a7e71d7..b743697 100755
--- a/src/mesh_u.c
+++ b/src/mesh_u.c
@@ -38,7 +38,7 @@
 
 #include <string.h>
 #ifndef _MSC_VER
-#include <values.h>
+#include <limits.h>
 #include <sys/time.h>
 #endif
 #include <time.h>
-- 
2.10.1

