diff -u -Nr -U5 silo-4.11-bsd/configure silo-4.11-bsd.szip-library-name/configure
--- silo-4.11-bsd/configure	2021-09-10 18:22:52.000000000 -0400
+++ silo-4.11-bsd.szip-library-name/configure	2023-09-04 13:04:48.458755404 -0400
@@ -26075,23 +26075,23 @@
 case $withval in
   default)
     HAVE_SZLIB="no"
     if test "$DEFAULT_SZIP" != "" ; then
       LDFLAGS="$LDFLAGS -L$DEFAULT_SZIP/lib"
-      LIBS="$LIBS -lsz"
+      LIBS="$LIBS -lszip"
       HAVE_SZLIB="yes"
       szlib_lib="$DEFAULT_SZIP/lib"
     fi
     if test "$DEFAULT_SZIP_LIB" != "" ; then
       LDFLAGS="$LDFLAGS -L$DEFAULT_SZIP_LIB"
-      LIBS="$LIBS -lsz"
+      LIBS="$LIBS -lszip"
       HAVE_SZLIB="yes"
       szlib_lib="$DEFAULT_SZIP_LIB"
     fi
     if test "$DEFAULT_SZIP_LIBDIR" != "" ; then
       LDFLAGS="$LDFLAGS -L$DEFAULT_SZIP_LIBDIR"
-      LIBS="$LIBS -lsz"
+      LIBS="$LIBS -lszip"
       HAVE_SZLIB="yes"
       szlib_lib="$DEFAULT_SZIP_LIBDIR"
     fi
     if test "$HAVE_SZLIB" = "yes" ; then
       { $as_echo "$as_me:${as_lineno-$LINENO}: checking default szlib path provided" >&5
@@ -26113,23 +26113,23 @@
     ;;
   yes)
     HAVE_SZLIB="no"
     if test "$DEFAULT_SZIP" != "" ; then
       LDFLAGS="$LDFLAGS -L$DEFAULT_SZIP/lib"
-      LIBS="$LIBS -lsz"
+      LIBS="$LIBS -lszip"
       HAVE_SZLIB="yes"
       szlib_lib="$DEFAULT_SZIP/lib"
     fi
     if test "$DEFAULT_SZIP_LIB" != "" ; then
       LDFLAGS="$LDFLAGS -L$DEFAULT_SZIP_LIB"
-      LIBS="$LIBS -lsz"
+      LIBS="$LIBS -lszip"
       HAVE_SZLIB="yes"
       szlib_lib="$DEFAULT_SZIP_LIB"
     fi
     if test "$DEFAULT_SZIP_LIBDIR" != "" ; then
       LDFLAGS="$LDFLAGS -L$DEFAULT_SZIP_LIBDIR"
-      LIBS="$LIBS -lsz"
+      LIBS="$LIBS -lszip"
       HAVE_SZLIB="yes"
       szlib_lib="$DEFAULT_SZIP_LIBDIR"
     fi
     if test "$HAVE_SZLIB" = "yes" ; then
       { $as_echo "$as_me:${as_lineno-$LINENO}: checking using szlib path provided" >&5
@@ -26171,17 +26171,17 @@
     esac
     if test -n "$szlib_lib"; then
       LDFLAGS="$LDFLAGS -L$szlib_lib"
     fi
 
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for SZ_BufftoBuffCompress in -lsz" >&5
-$as_echo_n "checking for SZ_BufftoBuffCompress in -lsz... " >&6; }
+    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for SZ_BufftoBuffCompress in -lszip" >&5
+$as_echo_n "checking for SZ_BufftoBuffCompress in -lszip... " >&6; }
 if ${ac_cv_lib_sz_SZ_BufftoBuffCompress+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lsz  $LIBS"
+LIBS="-lszip  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
 /* Override any GCC internal prototype to avoid an error.
    Use char because int might match the return type of a GCC
@@ -26220,11 +26220,11 @@
 if test "x$ac_cv_lib_sz_SZ_BufftoBuffCompress" = xyes; then :
   cat >>confdefs.h <<_ACEOF
 #define HAVE_LIBSZ 1
 _ACEOF
 
-  LIBS="-lsz $LIBS"
+  LIBS="-lszip $LIBS"
 
 else
   LDFLAGS="$saved_LDFLAGS"; unset HAVE_SZLIB
 fi
 
@@ -26382,11 +26382,11 @@
 done
 
       tmp_LIBS="$LIBS"
       if test -n "$szlib_lib"; then
 ###      LIBS="$hdf5_lib/libhdf5.a $LIBS"
-         LIBS="-lhdf5 -lsz $LIBS"
+         LIBS="-lhdf5 -lszip $LIBS"
       else
          LIBS="-lhdf5 $LIBS"
       fi
       ac_fn_c_check_func "$LINENO" "H5open" "ac_cv_func_H5open"
 if test "x$ac_cv_func_H5open" = xyes; then :
@@ -26616,11 +26616,11 @@
 fi
 
             if test "$notfound" = ""; then
                if test -n "$szlib_lib"; then
 ###               LIBS="$hdf5_lib/libhdf5.a $tmp_LIBS"
-                  LIBS="-lhdf5 -lsz $tmp_LIBS"
+                  LIBS="-lhdf5 -lszip $tmp_LIBS"
                else
                   LIBS="-lhdf5 $tmp_LIBS"
                fi
             fi
             if test -n "$notfound"; then
@@ -26643,13 +26643,13 @@
 fi # if test -n "$HDF5_DRV"
 
 # Check to see if the results of the above tests found the HDF5 libraries
 if test -n "$HDF5_DRV"; then
     # Short-term hack to deal with bogus code above (too complex)
-    # putting -lsz AHEAD of -lhdf5 in libs
+    # putting -lszip AHEAD of -lhdf5 in libs
     if test "$HAVE_SZLIB" = "yes"; then
-        LIBS="$LIBS -lsz"
+        LIBS="$LIBS -lszip"
     fi
 
 $as_echo "#define HAVE_HDF5_DRIVER 1" >>confdefs.h
 
 else
